trace/HTTP.md

```diff
@@ -0,0 +1,105 @@
+# HTTP
+
+This document explains tracing of HTTP requests with OpenCensus.
+
+## Spans
+
+Implementations MUST create a span for outgoing requests at the
+client and a span for incoming requests at the server.
+
+Span name is formatted as:
+
+* Sent.$host[:$port]/$path for outgoing requests.
+* Recv.$host[:$port]/$path for incoming requests.
+
+Port MUST be omitted if it is 80 or 443.
+
+Examples of span names:
+
+* Sent.example.com/users
+* Recv.example.com:78/messages
+* Sent.example.com/users/25f4c31d
+
+## Propagation
+
+Propagation is how SpanContext is transmitted on the wire
+in an HTTP request.
+
+Implementations MUST allow users to set their own propagation
+format and MUST provide an implementation for B3 at least.
+
+If user doesn't set any propagation methods explicitly, B3 is used.
+
+The propagation method SHOULD modify a request object to insert
+a SpanContext or SHOULD be able to extract a SpanContext from a
+request object.
+
+## Status
+
+Implementations should set status if HTTP request or response
+is not successful (not 2xx).
```

Comment:
what about 3xx if the HTTP client does not have an redirect mechanism, is that still counted as successful?

Reply:
IMHO, a 3xx should be considered successful in my opinion. Because it satisfies the redirect request. The user can explicitly set an error status if they are not satisfied with an HTTP 3xx. I will add a note about3xx here.